integration-tests/src/tests/client/features/delegate_action.rs

```diff
@@ -395,3 +416,148 @@ fn meta_tx_delete_account() {
+fn meta_tx_ft_transfer() {
+    let relayer = alice_account();
+    let sender = bob_account();
+    let ft_contract = carol_account();
+    let receiver = "david.near";
+
+    let mut genesis = Genesis::test(vec![alice_account(), bob_account(), carol_account()], 3);
+    add_contract(&mut genesis, &ft_contract, near_test_contracts::ft_contract().to_vec());
+    let node = RuntimeNode::new_from_genesis(&relayer, genesis);
+
+    // A BUNCH OF TEST SETUP
+    // initialize the contract
+    node.user()
+        .function_call(
+            relayer.clone(),
+            ft_contract.clone(),
+            "new_default_meta",
+            // make the relayer (alice) owner, makes initialization easier
+            br#"{"owner_id": "alice.near", "total_supply": "1000000"}"#.to_vec(),
+            30_000_000_000_000,
+            0,
+        )
+        .expect("FT contract initialization failed")
+        .assert_success();
+
+    // register sender & receiver FT accounts
+    let actions = vec![ft_register_action(&sender), ft_register_action(&receiver)];
+    node.user()
+        .sign_and_commit_actions(relayer.clone(), ft_contract.clone(), actions)
+        .expect("registering FT accounts")
+        .assert_success();
+    // initialize sender balance
+    let actions = vec![ft_transfer_action(&sender, 10_000).0];
+    node.user()
+        .sign_and_commit_actions(relayer.clone(), ft_contract.clone(), actions)
+        .expect("initializing sender balance failed")
+        .assert_success();
+
+    // START OF META TRANSACTION
+    // 1% fee to the relayer
+    let (action0, bytes0) = ft_transfer_action(&relayer, 10);
+    // the actual transfer
+    let (action1, bytes1) = ft_transfer_action(receiver, 1000);
+    let actions = vec![action0, action1];
+
+    let outcome = check_meta_tx_fn_call(
+        &node,
+        actions,
+        bytes0 + bytes1,
+        2,
+        sender.clone(),
+        relayer.clone(),
+        ft_contract,
+    );
+
+    // Check that the function call was executed as expected, according to NEP-141 events.
+    let fn_call_logs = &outcome.receipts_outcome[1].outcome.logs;
+
+    assert_eq!(2, fn_call_logs.len(), "expected 2 JSON events but found {fn_call_logs:?}");
+    assert_eq!(
+        fn_call_logs[0],
+        ft_transfer_event(&sender, &relayer, 10),
+        "relayer event looks wrong"
+    );
+    assert_eq!(
+        fn_call_logs[1],
+        ft_transfer_event(&sender, &receiver, 1000),
+        "receiver event looks wrong"
+    );
+
+    // let relayer_payment_event: serde_json::Value =
+    //     serde_json::from_str(&format!("{{{}}}", fn_call_logs[0]))
+    //         .expect("log events should be JSON");
+    // assert_eq!(relayer_payment_event, ft_transfer_event(&sender, &relayer, 10));
+
+    // let transfer_event: serde_json::Value =
+    //     serde_json::from_str(&fn_call_logs[1]).expect("log events should be JSON");
+    // assert_eq!(transfer_event, ft_transfer_event(&sender, &receiver, 1000));
```

Comment:
can you also check that david got the tokens ? (and that the token balance of alice & bob are correct)

Reply:
yes, I added a small helper `assert_ft_balance` and then 

```rust
assert_ft_balance(&node, &ft_contract, &receiver, 1000);
assert_ft_balance(&node, &ft_contract, &sender, 10_000 - 1000 - 10);
assert_ft_balance(&node, &ft_contract, &relayer, 1_000_000 - 10_000 + 10);
```